asyncpg/connection.py

```diff
@@ -2087,6 +2099,15 @@ async def connect(dsn=None, *,
     if record_class is not protocol.Record:
         _check_record_class(record_class)
 
+    try:
+        target_session_attribute = SessionAttribute(target_session_attribute)
```

Comment:
Please move this check to `_parse_connect_dsn_and_args`.  That is also the place where the `PGTARGETSESSIONATTRS` can be handled in case an explicit argument wasn't passed to `connect()`.